### PR TITLE
TRA-3860 create main configmap for agent and tappers

### PR DESCRIPTION
**Pull request by RamiBerm**
Mostly a placeholder now, will be more meaningful once daemon mode is fully implemented

### CODE REVIEW

File: agent/pkg/config/config.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+	defaultRegexTarget          string = ".*"
+)
+
+var Config *shared.MizuAgentConfig
+
+func LoadConfig() error {
+	if Config != nil {
+		return nil
+	}
+	filePath := fmt.Sprintf("%s%s", shared.ConfigDirPath, shared.ConfigFileName)
```
Comment (RamiBerm) [inline agent/pkg/config/config.go:24]: Use file not exist func from cli

Potentially move read json file to shared

File: cli/cmd/tapRunner.go [hunk 2]
```diff
@@ -54,24 +55,24 @@ func RunMizuTap() {
 		return
 	}
 
-	var mizuValidationRules string
+	var serializedRulePolicy string
```
Comment (RoyUP9) [inline cli/cmd/tapRunner.go:58]: rename rule policy to validation rules

File: agent/pkg/config/config.go [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+	content, err := ioutil.ReadFile(filePath)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return applyDefaultConfig()
+		}
+		return err
+	}
+	err = json.Unmarshal(content, &Config)
```
Comment (RoyUP9) [inline agent/pkg/config/config.go:32]: change to one liner

File: cli/config/config.go [hunk 4]
```diff
@@ -363,3 +364,27 @@ func setZeroForReadonlyFields(currentElem reflect.Value) {
...
+	serializableRegex, err := shared.CompileRegexToSerializableRegexp(Config.Tap.PodRegexStr)
+	if err != nil {
+		return nil, err
+	}
+	config := shared.MizuAgentConfig{
+		TapTargetRegex: *serializableRegex,
+		MaxDBSizeBytes: Config.Tap.MaxEntriesDBSizeBytes(),
+	}
+	return &config, nil
+}
```
Comment (RoyUP9) [inline cli/config/config.go:390]: add new line

**Review by RoyUP9 — APPROVED** [review]

**Pull request merged**